tox.ini

```
@@ -9,19 +9,22 @@ python =
 DJANGO =
     3.2: dj32
     4.0: dj40
+    4.1: dj41
 
 WAGTAIL =
     2.15: wt215
     2.16: wt216
     3.0: wt30
+    4.0: wt40
 
 [tox]
 skipsdist = True
 usedevelop = True
 
 envlist =
-    py{37,38,39,310}-dj32-wt{215,216,30}
-    py{38,39,310}-dj40-wt{216,30}
+    py{37,38,39,310}-dj32-wt{215,216,30,40}
+    py{38,39,310}-dj40-wt{216,30,40}
+    py{38,39,310}-dj41-wt{216,30,40}
```

You can simplify the `test.yml` by making this modification since only Wagtail 4 is compatible with Django 4.1.

```suggestion
    py{38,39,310}-dj41-wt40
```

---

There _are_ still some compatibility errors, though